packages/zowe-explorer/src/Profiles.ts

```diff
@@ -1103,6 +1121,154 @@ export class Profiles extends ProfilesCache {
         return updatedServiceProfile;
     }
 
+    public async ssoLogin(node?: IZoweNodeType, label?: string) {
+        const baseProfile = await this.getBaseProfile();
+        let serviceProfile: IProfileLoaded;
+        if (node) {
+            serviceProfile = node.getProfile();
+        } else {
+            serviceProfile = this.loadNamedProfile(label.trim());
+        }
+
+        // This check will handle service profiles that have username and password
+        if (serviceProfile.profile.user && serviceProfile.profile.password) {
+            vscode.window.showInformationMessage(
+                localize("ssoLogin.noBase", "Log in skipped. This profile does not support base profiles.")
+            );
+            return;
+        }
+
+        // This check is for optional credentials
+        if (
+            baseProfile &&
+            serviceProfile.profile.host &&
+            serviceProfile.profile.port &&
+            ((baseProfile.profile.host !== serviceProfile.profile.host &&
+                baseProfile.profile.port !== serviceProfile.profile.port) ||
+                (baseProfile.profile.host === serviceProfile.profile.host &&
+                    baseProfile.profile.port !== serviceProfile.profile.port))
+        ) {
+            vscode.window.showInformationMessage(
+                localize("ssoLogin.noBase", "Log in skipped. This profile does not support base profiles.")
+            );
+            return;
+        }
+
+        let newUser: string;
+        let newPass: string;
+
+        if (baseProfile) {
+            newUser = await this.userInfo();
+            if (newUser === undefined) {
+                vscode.window.showInformationMessage(localize("ssoLogin.undefined.username", "Operation Cancelled"));
+                return;
+            } else {
+                newPass = await this.passwordInfo();
+                if (newPass === undefined) {
+                    vscode.window.showInformationMessage(
+                        localize("ssoLogin.undefined.username", "Operation Cancelled")
+                    );
+                    return;
+                }
+            }
+
+            try {
+                const combinedProfile = await Profiles.getInstance().getCombinedProfile(serviceProfile, baseProfile);
+                const updSession = new Session({
+                    hostname: combinedProfile.profile.host,
+                    port: combinedProfile.profile.port,
+                    user: newUser,
+                    password: newPass,
+                    rejectUnauthorized: combinedProfile.profile.rejectUnauthorized,
+                    tokenType: combinedProfile.profile.tokenType,
+                    type: combinedProfile.profile.type,
+                });
+                const loginToken = await zowe.Login.apimlLogin(updSession);
```

Comment:
This needs to be added to the Common API as we cannot call zOSMF specific implementations in Zowe Explorer code.